### PR TITLE
feat: Client emits events for log{in,out}

### PR DESCRIPTION
**Pull request by ptbrowne**
Emitting events during login/logout provides a flexible way
for components/objects using CozyClient to setup/destroy
themselves. It lets the setup to be decoupled from where
the login/logout happens.

I think it can be a good way for example to init the bar.

```
cozyClient.on('login', setupBar)
cozyClient.on('logout', destroyBar)
```

I would like to have your views on this.

### CODE REVIEW

**Comment by Crash--** [comment]
I'm a big fan on an event solution, but it'll bring a new concept in Cozy-Client and I don't know if I like the idea. 

We already have `callback` concept in Cozy-Client, specially in `OAuthClient` with the `onTokenRefresh`. Is it something we can follow for this ?

```jsx 
const client = this.context.client
client.onLogin = () => setupBar()
```

**Comment by drazik** [comment]
I like it, but I agree with @Crash--, we must go full events or full callbacks but mixing the two is weird. IMHO, events are great, I like your example.

**Comment by ptbrowne** [comment]
Yes you are both right, having both is not good.

I would prefer to go full events since with the callback method you need to have all objects used in the callback ready at client instantation.

**Comment by Crash--** [comment]
> I don't mind having both at the same time if it's a temporary thing.
Nothing is temporary :( 

Except the `onTokenResfresh` where do we use callbacks?

**Comment by ptbrowne** [comment]
> Nothing is temporary :(

Yes but we can add deprecation warning since we will not be able to remove onTokenRefresh right away.

I think the best here would be to deprecate onTokenRefresh and add an event tokenRefreshed ?

Then we could merge ? WDYT ?

**Comment by ptbrowne** [comment]
I'd be happy to hear what @gregorylegarec and @CPatchane think about this since the bar would be one of the primary consumer of login/logout events.

**Comment by kosssi** [comment]
👍 I talk with @ptbrowne to add event on `onTokenRefresh` but I never see this PR ^^

I would like use this function to fix realtime on mobile app when token is not good.

File: packages/cozy-client/src/CozyClient.js [hunk 1]
```diff
@@ -690,9 +690,21 @@ class CozyClient {
     if (stackClient) {
       this.stackClient = stackClient
     } else {
+      const options = {
+        ...this.options,
+        onTokenRefresh: token => {
+          this.emit('tokenUpdated', token)
```
Comment (ptbrowne) [inline packages/cozy-client/src/CozyClient.js]: s/tokenUpdated/tokenRefreshed

Let's not introduce new words so that it is easier to transition ;)

Comment (ptbrowne) [reply packages/cozy-client/src/CozyClient.js]: I think we should make it harder to get the token as it should be internal to cozyClient.

Something like `this.emit('tokenRefreshed', { dangerousToken: token })` ?

File: packages/cozy-client/src/CozyClient.js [hunk 2]
```diff
@@ -690,9 +690,21 @@ class CozyClient {
...
+          if (this.options.onTokenRefresh) {
+            deprecatedHandler(
+              `Using onTokenRefresh is deprecated, please use events like this: cozyClient.on('tokenUpdated', token => console.log('Token is updated', token)). https://git.io/fj3M3`
+            )
```
Comment (ptbrowne) [inline packages/cozy-client/src/CozyClient.js]: 👍

File: packages/cozy-client/src/CozyClient.js (hunk 1 above)
Comment (kosssi) [reply packages/cozy-client/src/CozyClient.js]: I remove token to force the developer to go through the cozyClient to retrieve the token if he needs it.

**Pull request merged**